packages/utilities/crypto/lib/types.web.ts

```diff
@@ -0,0 +1,32 @@
+/**
+ * Represents a public cryptographic key
+ */
+export interface PublicKey {
+	/** Returns the public key bytes */
+	bytes(): Promise<Uint8Array>;
+	/** Returns a did:key representation of the public key */
+	did(): Promise<`did:key:${string}`>;
+	/** Verifies a signature against a provided data */
+	verify(sig: Uint8Array, data: Uint8Array, options?: VerifyOptions): Promise<boolean>;
+}
+
+/**
+ * Represents a private cryptographic key
+ */
+export interface PrivateKey extends PublicKey {
+	/** Signs provided data using the private key */
+	sign(data: Uint8Array): Promise<Uint8Array>;
+}
+
+/**
+ * Represents an exportable private cryptographic key
+ */
+export interface PrivateKeyExportable extends PrivateKey {
+	/** Exports the private key */
+	export(type: 'hex' | 'multikey'): Promise<string>;
+	export(type: 'bytes'): Promise<Uint8Array>;
```

Comment:
likewise `hex`, `bytes`, aren't very specific. `raw_hex` and `raw_bytes` maybe? (I know "raw" isn't a format either but I think it's descriptive enough)

Reply:
fwiw, `raw` would be consistent with the `raw` format of WebCrypto (even if it doesn't support it for private keys natively).

Reply:
I revamped the api in 9d6673bbf09fbf17593375a9cdcef7f10d9b8ee5, let me know what you think